### DISCUSSION TITLE
Break line reconnection

### DISCUSSION THREAD

**Discussion by 1136386803** [bepu/bepuphysics1#33] [Q&A]
Use "Controller.HorizontalMotionConstraint.MovementDirection"  Unable to follow frame position

**Comment by 1136386803** [comment]
”CharacterController.HorizontalMotionConstraint.MovementDirection” How do I predict the position of the next frame

**Comment by RossNordby** [comment]
The movement direction is in terms of a 2D basis where the +Y is along the `HorizontalMotionConstraint.ForwardDirection` and +X is along the `HorizontalMotionConstraint.StrafeDirection`. Those directions are created from the character's view direction. `HorizontalMotionConstraint.MovementDirection3d` is the direction in 3d space that the character will try to move in.

Fully predicting the next location is more complicated because it requires being aware of the support and any obstacles. It would basically requiring simulation.

(I'm not sure if I'm actually answering your question, I don't quite understand what you're looking for.)
